src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -87,20 +118,60 @@ private static bool ApplyChangesFromTC25ThroughTC28(JObject sarifLog)
                         modifiedLog = true;
                     }
 
+                    // Logical locations are now an array. We will first transform the previous
+                    // dictionary into the arrays form. We will retain the index for each
+                    // transformed logical location. Later, we will associate these indices
+                    // with results, if applicable.
+                    var logicalLocations = run["logicalLocations"] as JObject;
+                    Dictionary<LogicalLocation, int> logicalLocationToIndexMap = null;
+
+                    if (logicalLocations != null)
+                    {
+                        logicalLocationToIndexMap = new Dictionary<LogicalLocation, int>(LogicalLocation.ValueComparer);
+                        var jObjectToIndexMap = new Dictionary<JObject, int>();
+
+                        run["logicalLocations"] =
+                            ConstructLogicalLocationsArray(
+                                logicalLocations,
+                                jObjectToIndexMap,
+                                logicalLocationToIndexMap,
```

Comment:
>logicalLocationToIndexMap [](start = 32, length = 25)

BUG: We pass this empty dictionary into `ConstructLogicalLocationsArray`. But that function immediately overwrites the parameter with a new empty dictionary. So the passed-in dictionary is never modified, and you always pass the empty dictionary to the `LogicalLocationIndexRemapper` on Line 164.

Reply:
This has got to indicate a test hole. I don't see how the call to `Visit` on Line 166 can work if the remapper always holds an empty array.

---
In reply to: [243855820](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243855820) [](ancestors = 243855820)